packages/connectkit/src/hooks/usePayWithSolanaToken.ts

```diff
@@ -56,10 +63,12 @@ export function usePayWithSolanaToken({
       }
     })();
 
+    // TOOD: get the actual amount sent from the tx logs.
+    // We are currently using a fake amount = 0.
     trpc.processSolanaSourcePayment.mutate({
       orderId: orderId.toString(),
       startIntentTxHash: txHash,
-      amount: chosenFinalTokenAmount,
+      amount: "0", // TODO: replace.
```

Comment:
fix

Reply:
what's the best way to get the amount of input token sent from the `txHash` (not sure how receipts work on Solana... do they have something like the Transfer event log we can fetch?)

- we are using an exact-output swap, so we don't know how much input token will be consumed upfront, only once it's onchain
- the previous code was substituting a totally different amount (different token, different decimals) as far as i can tell